channeldb/payment_control.go

```diff
@@ -362,8 +410,22 @@ func nextPaymentSequence(tx *bbolt.Tx) ([]byte, error) {
 // fetchPaymentStatus fetches the payment status of the payment. If the payment
 // isn't found, it will default to "StatusUnknown".
 func fetchPaymentStatus(bucket *bbolt.Bucket) (PaymentStatus, error) {
-	if bucket.Get(paymentSettleInfoKey) != nil {
-		return StatusSucceeded, nil
+	htlcsBucket := bucket.Bucket(paymentHtlcsBucket)
+	if htlcsBucket != nil {
+		htlcs, err := fetchHTLCAttempts(htlcsBucket)
+		if err != nil {
+			return 0, err
+		}
+
+		// Go through all HTLCs, and return StatusSucceeded if any of
```

Comment:
Why is it a success if any one of them succeed? Shouldn't they all need to succeed for the entire payment to be considered a success?

Reply:
Our current thought is that having the preimage means that the payment succeeded. But this can be detailed further in the mpp pr.